dynamic-plugin/src/components/NGUI/ResourceListWidget.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useMemo } from 'react';
+import {
+  K8sResourceCommon,
+  RowProps,
+  TableColumn,
+  TableData,
+  VirtualizedTable,
+  useK8sWatchResource,
+  useK8sModel,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { ColumnSpec, DataTypeConfig } from './dataTypeRegistry';
+import {
+  getByPath,
+  getSchemaFormat,
+  matchFieldToColumnSpec,
+  NGUIField,
+  typeRenderers,
+} from './widget-utils';
+
+type ResourceListWidgetProps = {
+  config?: DataTypeConfig;
+  fields: NGUIField[];
+  k8s?: {
+    group?: string;
+    version?: string;
+    kind?: string;
+    namespaced?: boolean;
+    namespace?: string;
+  };
+};
+
+const ResourceListWidget: React.FC<ResourceListWidgetProps> = ({ config, fields, k8s }) => {
+  const groupVersionKind = {
+    group: k8s?.group,
+    version: k8s?.version,
+    kind: k8s?.kind,
+  };
+
+  const [items, loaded, loadError] = useK8sWatchResource<K8sResourceCommon[]>({
+    groupVersionKind,
+    isList: true,
+    namespaced: k8s?.namespaced,
+    namespace: k8s?.namespace,
+  });
+
+  const [model] = useK8sModel(groupVersionKind);
+
+  // TODO: Why am I not seeing a schema in the model?
+  console.log('model', model);
```

Comment:
Is this the correct way of pulling the schema to find the correct format for the data we're trying to render? I'm not getting anything back for `Pod`...does it have to be a custom resource?

Reply:
@jhadvig Looks like we don't expose any of the swagger / OpenAPI utilities in the SDK. Something we should think about adding? We'll likely need it for Genie.

https://github.com/openshift/console/blob/main/frontend/public/module/k8s/swagger.ts

Reply:
We can temporarily get the schema from `localStorage` using this key if needed, but it would be best if we can expose the utilities in the SDK:

https://github.com/openshift/console/blob/main/frontend/public/module/k8s/swagger.ts#L8